### PR TITLE
ignore guzzlehttp/guzzle

### PR DESCRIPTION
**Pull request by ArtificialOwl**
ignore `guzzlehttp/guzzle` to avoid conflict with the one from core

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -17,7 +17,6 @@
         "nextcloud/coding-standard": "^1",
         "psalm/phar": "^4.3",
         "nextcloud/ocp": "dev-master",
-        "guzzlehttp/guzzle": "^7.5.1",
```
Comment (marcelklehr) [inline composer.json:20]: Mmh, this is require-dev, though. It shouldn't be shipped, no?

File: composer.json [hunk 2]
```diff
@@ -27,5 +26,8 @@
         "allow-plugins": {
             "php-http/discovery": true
         }
+    },
+    "replace": {
+        "guzzlehttp/guzzle": "*"
```
Comment (marcelklehr) [inline composer.json:31]: This would mean integration_onedrive replaces guzzle?

**Review by marcelklehr — CHANGES_REQUESTED** [review]
Guzzle is currently not shipped as part of this app, AFAIK. I think you're on the wrong track.

File: composer.json (hunk 1 above)
Comment (marcelklehr) [reply composer.json:20]: ok, I see kiota requires guzzle

Still, can you please keep in the require-dev for guzzle, so our psalm tests work?

**Comment by ArtificialOwl** [comment]
> Guzzle is currently not shipped as part of this app, AFAIK. I think you're on the wrong track.

guzzle is shipped as dependency of `microsoft/kiota-http-guzzle`
https://github.com/nextcloud/integration_onedrive/blob/db1304805787584d10ee5fd09cb7a751c7946496/composer.lock#L317C11-L336C15

File: composer.json (hunk 2 above)
Comment (bigcat88) [reply composer.json:32]: I tried honestly different ways to fix this CI locally, but with `"guzzlehttp/guzzle"` in `replace` we need to manually put Guzzle into the CI - I can add it to this PR, but I'm not sure that it's correct.

**Review by marcelklehr — APPROVED** [review]

**Pull request merged**